### ISSUE TITLE
rows_patch fails when y has key values that don't occur in x

### ISSUE THREAD

**Issue by melissagwolf** [tidyverse/dplyr#5984]
I tried to use `row_patch` to fill in missing values in a data frame.  It failed by design, because it didn't meet the requirement for the function: "key values in y must occur in x".  However, I was able to successfully execute the row patch by using the `match` function from base R (see below).  

This isn't a typical bug report because it is clearly stated that this is a limitation of the function, but I wanted to point out that it can be done using a different function from base R.  Since `row_patch` is currently an experimental function in `dplyr`, I wanted to provide feedback to hopefully improve it.  `row_patch` is much cleaner and I would prefer to use it if possible.

It would also be nice if `row_patch` could preserve other variables in y that don't occur in x.  `match` can handle this as well.

```r
D1 <- data.frame(
  id=seq(1,3),
  x=c("cow",NA,"sheep"))

D2 <- data.frame(
  id=seq(1,4),
  x=c("cow","turtle","parrot","frog"))

#rows_patch fails
D1 %>%
  rows_patch(D2, by = "id")

#match works
na <- is.na(D1$x)

D1$x[na] <- D2$x[match(D1$id[na],D2$id)]
```

**Comment by hadley** [comment]
Reprex:

``` r
library(dplyr, warn.conflicts = FALSE)
D1 <- data.frame(
  id = seq(1, 3),
  x = c("cow", NA, "sheep")
)

D2 <- data.frame(
  id = seq(1, 4),
  x = c("cow", "turtle", "parrot", "frog")
)

# rows_patch fails
D1 %>%
  rows_patch(D2, by = "id")
#> Error: Attempting to patch missing rows.

# match works
na <- is.na(D1$x)

D1$x[na] <- D2$x[match(D1$id[na], D2$id)]
```

<sup>Created on 2021-09-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by hadley** [comment]
Might be worthwhile to have @DavisVaughan think about these functions once he's done with his current work on joins.

**Comment by DavisVaughan** [comment]
Probably would be fixed by #5588 if this comment is still applicable in that PR https://github.com/tidyverse/dplyr/pull/5588#issuecomment-722522654

> I think is makes sense that keys in y that do not exist in x should always be silently ignored when updating/deleting/patching (no changes needed - we do this in this PR).

And @mgirlich followed up that it might be useful to make that an optional check, rather than just silently ignoring them https://github.com/tidyverse/dplyr/pull/5588#issuecomment-918861336
